test/CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+cmake_minimum_required ( VERSION 3.13.0 )
+project ( "CoreMQTT unit test"
+          VERSION 1.0.0
+          LANGUAGES C )
+
+# Allow the project to be organized into folders.
+set_property( GLOBAL PROPERTY USE_FOLDERS ON )
+
+# Use C90.
+set( CMAKE_C_STANDARD 90 )
+set( CMAKE_C_STANDARD_REQUIRED ON )
+
+# Do not allow in-source build.
+if( ${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR} )
+    message( FATAL_ERROR "In-source build is not allowed. Please build in a separate directory, such as ${PROJECT_SOURCE_DIR}/build." )
+endif()
+
+# Set global path variables.
+get_filename_component(__MODULE_ROOT_DIR "${CMAKE_CURRENT_LIST_DIR}/.." ABSOLUTE)
+set(MODULE_ROOT_DIR ${__MODULE_ROOT_DIR} CACHE INTERNAL "coreMQTT repository root.")
+
+# Configure options to always show in CMake GUI.
+option( BUILD_CLONE_SUBMODULES
+        "Set this to ON to automatically clone any required Git submodules. When OFF, submodules must be manually cloned."
+        ON )
+
+# Set output directories.
+set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
+set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
+set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
+
+# ===================================== Library Configuration =================================================
+
+# Include filepaths for source and include.
+include( ${MODULE_ROOT_DIR}/mqttFilePaths.cmake )
+
+# MQTT library target.
+add_library( core_mqtt
```

Comment:
I have spoken with Nathan and sent him all of the commands to run coverity using the unit test target as it is. The only thing that would use extra target is coverity, so I do not think an extra target is necessary. This extra target does not help with the advisory 8.7 and 2.3 rules.

Reply:
The extra target enables us to test the `MQTT_DO_NOT_USE_CUSTOM_CONFIG` macro, which cannot be tested through the unit test target which requires use of a `core_mqtt_config.h` file to provide definition of `NetworkContext_t`

Reply:
Ok I can't think of any other way to test other than the CSDK demos